imap_processing/hit/l0/decom_hit.py

```diff
@@ -0,0 +1,484 @@
+    sequential_check = [is_sequential(counters[idx : idx + size]) for idx in indices]
+    return indices[sequential_check]
+
+
+def update_ccsds_header_dims(sci_dataset: xr.Dataset) -> xr.Dataset:
+    """
+    Update dimensions of CCSDS header fields.
+
+    The CCSDS header fields contain 1D arrays with
+    values from all the packets in the file.
+    While the epoch dimension contains time per packet,
+    it will be updated later in the process to represent
+    time per science frame, so another time dimension is
+    needed for the ccsds header fields.This function
+    updates the dimension for these fields to use sc_tick
+    instead of epoch. sc_tick is the time the packet was
+    created.
+
+    Parameters
+    ----------
+    sci_dataset : xr.Dataset
+        Xarray dataset containing HIT science packets
+        from a CCSDS file.
+
+    Returns
+    -------
+    sci_dataset : xr.Dataset
+        Updated xarray dataset.
+    """
+    # sc_tick contains spacecraft time per packet
+    sci_dataset.coords["sc_tick"] = sci_dataset["sc_tick"]
+    sci_dataset = sci_dataset.swap_dims({"epoch": "sc_tick"})
+    return sci_dataset
+
+
+def assemble_science_frames(sci_dataset: xr.Dataset) -> xr.Dataset:
+    """
+    Group packets into science frames.
+
+    HIT science frames (data from 1 minute) consist of 20 packets.
+    These are assembled from the binary science_data field in the
+    xarray dataset, which is a 1D array of science data from all
+    packets in the file, by using packet grouping flags.
+
+    The science frame is further categorized into
+    L1A data products -> count rates and event data.
+
+        The first six packets contain count rates data
+        The last 14 packets contain pulse height event data
+
+    These groups are added to the dataset as count_rates_binary
+    and pha_binary.
+
+    Parameters
+    ----------
+    sci_dataset : xr.Dataset
+        Xarray Dataset for science data (APID 1252).
+
+    Returns
+    -------
+    sci_dataset : xr.Dataset
+        Updated xarray dataset with binary count rates and pulse
+        height event data per valid science frame added as new
+        data variables.
+    """
+    # TODO: Figure out how to handle partial science frames at the
+    #  beginning and end of CCSDS files. These science frames are split
+    #  across CCSDS files and still need to be processed with packets
+    #  from the previous file. Only discard incomplete science frames
+    #  in the middle of the CCSDS file. The code currently skips all
+    #  incomplete science frames.
+
+    # Convert sequence flags and counters to NumPy arrays for vectorized operations
+    seq_flgs = sci_dataset.seq_flgs.values
+    seq_ctrs = sci_dataset.src_seq_ctr.values
+    science_data = sci_dataset.science_data.values
+    epoch_data = sci_dataset.epoch.values
+
+    # Number of packets in the file
+    total_packets = len(epoch_data)
+    frame_size = 20
```

Comment:
This is just an idea that I'm not sure is really better, so feel free to ignore it: you could make the flag pattern a global variable and use the length of that to get the frame_size instead of hardcoding that. Another option is to pull the APID Enum out into a hit config type file and the flag pattern could be added as a variable in that file. The flag pattern could then be imported into this file.

Reply:
I think global variable could be helpful. Then you can use that global variable in `line 246` and `line 283` since they should be equal.

Reply:
I like this suggestion! I've added a global variable for now and a TODO to consider moving it to a config file in the future